extensions/iceberg/src/main/java/io/deephaven/iceberg/location/IcebergTableLocationKey.java

```diff
@@ -0,0 +1,70 @@
+//
+// Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.iceberg.location;
+
+import io.deephaven.engine.table.impl.locations.TableLocationKey;
+import io.deephaven.engine.table.impl.locations.local.URITableLocationKey;
+import io.deephaven.parquet.table.ParquetInstructions;
+import io.deephaven.parquet.table.location.ParquetTableLocationKey;
+import org.apache.iceberg.FileFormat;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.net.URI;
+import java.util.Map;
+
+/**
+ * {@link TableLocationKey} implementation for use with data stored in the parquet format.
+ */
+public class IcebergTableLocationKey extends URITableLocationKey {
```

Comment:
I think you make a strong argument for having Parquet (and later ORC) specific subclasses.
_OR_
We could just operate on `URITableLocationKey`
_OR_
Make `IcebergTableLocationKey` a marker sub-interface of `TableLocationKey`, and then have `IcebergParquetTableLocationKey extends URITableLocationKey implements IcebergTableLocationKey`, and similar for ORC.